app/models/inventory-batch.js

```diff
@@ -16,7 +16,7 @@ export default AbstractModel.extend({
     },
     inventoryItemTypeAhead: {
       presence: {
-        if(object) {
+        unless(object) {
```

Comment:
This doesn't resolve the issue.  ```unless(object)``` isn't the same as ```unless(object.haveInvoiceItems)```
The validation needs to fire when object.haveInvoiceItems is false.